content/docs/quickstart/cpp.md

```diff
@@ -3,69 +3,163 @@ layout: quickstart
+cmake-version: 3.17.0
 ---
 
-### Prerequisites
+In the C++ world, there's no universally accepted standard for managing project
+dependencies. In this quick start, you'll follow steps to build and locally
+install gRPC before building and running this quick start's Hello World example.
+
+### Setup
 
-#### gRPC
+Choose a directory to hold locally installed packages. This page assumes that
+the environment variable `MY_INSTALL_DIR` holds this directory path. For
+example:
 
-To install gRPC on your system, follow the [instructions to install gRPC C++ via make](https://github.com/grpc/grpc/blob/master/src/cpp/README.md#make).
+```sh
+$ export MY_INSTALL_DIR=$HOME/local
+```
 
-To run the example code, please ensure `pkg-config` is installed on your
-machine before you build and install gRPC in the previous step, since the
-example `Makefile`s try to look up the installed gRPC path using `pkg-config`.
-On Debian-based systems like Ubuntu, this can usually be done via
-`sudo apt-get install pkg-config`.
+Ensure that the directory exists:
 
-#### Protocol Buffers v3
+```sh
+$ mkdir -p $MY_INSTALL_DIR
+```
 
-While not mandatory to use gRPC, gRPC applications usually leverage Protocol
-Buffers v3 for service definitions and data serialization, and our example code
-uses Protocol Buffers as well as gRPC. If you don't already have it installed on
-your system, you can install the version cloned alongside gRPC. First ensure
-that you are running these commands in the gRPC tree you just built in the from
-the previous step.
+Add the local `bin` folder to your path variable, for example:
 
 ```sh
-$ cd third_party/protobuf
-$ make && sudo make install
+$ export PATH="$PATH:$MY_INSTALL_DIR/bin"
 ```
 
-###  Build the example
+### Prerequisites
 
-Always assuming you have gRPC properly installed, go into the example's
-directory:
+#### cmake
+
+Version 3.13 or later of `cmake` is required to install gRPC locally.
+
+- Linux
+
+  ```sh
+  $ sudo apt install -y cmake
+  ```
+
+- macOS:
+
+  ```sh
+  $ brew install cmake
+  ```
+
+- For general `cmake` installation instructions, see [Installing CMake][].
+
+Check the version of `cmake`:
 
 ```sh
-$ cd examples/cpp/helloworld/
+$ cmake --version
 ```
 
-Let's build the example client and server:
+Under Linux, the version of the system-wide `cmake` can be too low. You can
+install a more recent version into your local installation directory as follows:
+
 ```sh
-$ make
+$ wget -q -O cmake-linux.sh https://github.com/Kitware/CMake/releases/download/v{{< param cmake-version >}}/cmake-{{< param cmake-version >}}-Linux-x86_64.sh
+$ sh cmake-linux.sh -- --skip-license --prefix=$MY_INSTALL_DIR
+$ rm cmake-linux.sh
 ```
 
-Most failures at this point are a result of a faulty installation (or having
-installed gRPC to a non-standard location. Check out [the installation
-instructions for details](https://github.com/grpc/grpc/blob/master/src/cpp/README.md#make)).
+
+#### gRPC and Protocol Buffers v3
```

Comment:
note that protobuf 3.x.y (which is what some people will understand under protocol buffers v3) and "proto3" are not the same thing.   First is the version of protobuf release and  "proto3" is the new syntax for .proto files.

Given that proto3 support has been around for several years now, I think it's fine to omit the "v3" from the section title. (maybe add a note that the recommended syntax is proto3).

Reply:
Done